### ISSUE TITLE
RFE: add DefaultSession= setting

### ISSUE THREAD

**Issue by kkofler** [sddm/sddm#326]
I would like to request a DefaultSession= setting in sddm.conf, which decides which section to preselect when there is no remembered session. Currently, such a setting exists only for auto-login, but what session we default to for manual login also matters to users. In particular, in Fedora, we found that SDDM was defaulting to "KDE Plasma (Failsafe)" instead of the regular "KDE Plasma". (We're now working around that by renaming the failsafe session's .desktop file to come later in the alphabet, but that does not work in other setups, such as when there are multiple desktop environments, and the distribution or the local sysadmin want a specific one to be the default.) RememberLastSession is optional, and even when it's enabled (as it is on Fedora), there's always a first boot, and the default should be sane there.

**Comment by jleclanche** [comment]
Marking this as dupe of #346

**Comment by kkofler** [comment]
No, this is not a duplicate. #346 is about setting the preferred session *per user*. This RFE is about setting the preferred session *systemwide*.

**Comment by jleclanche** [comment]
It should be dealt with alongside regardless.

**Comment by jleclanche** [comment]
For what it's worth we should be reusing the same setting as the autologin one.

**Comment by kkofler** [comment]
I'd also like to point out that adding support for a setting like this is trivial, much easier than e.g. adding support for AccountsService as is being envisioned for #346.

**Comment by plfiorini** [comment]
@jleclanche No reusing the autologin setting is confusing plus it's empty by default.
This deserves a new setting.

**Comment by plfiorini** [comment]
Also, removing 'duplicate', this is a different feature request..

**Comment by tpgxyz** [comment]
Up!

**Comment by alienbob** [comment]
You have my vote too for adding a DefaultSession setting in sddm.conf.
The current implementation of just picking the first one you can find in an alphabetical listing of the 'xsessions' directory is not helpful.

**Comment by kkofler** [comment]
Almost 5 years (!) have passed and this feature, which is essential for distributions shipping more than one desktop environment, and which I expect to take less than a day to implement, is still missing.

**Comment by plfiorini** [comment]
@kkofler I'm sure that with your experience you can allocate a day to implement this feature, and send a patch.
I will be glad to review it, or if I'm too busy with my day job there are @davidedmundson @Vogtinator, you should know them very well from KDE.

**Comment by Vogtinator** [comment]
The way we use sddm in openSUSE makes this unnecessary for us - we use a symlink "default.desktop" using update-alternatives and its name is hardcoded in sddm itself.

This feature would allow dropping a (one-line) patch though, so there's indeed a benefit. Implementing this feature should indeed be trivial to do.

**Comment by PureTryOut** [comment]
@Vogtinator do you have a link to the patch by any chance?

**Comment by Vogtinator** [comment]
> @Vogtinator do you have a link to the patch by any chance?

It changes the default value of `[Last] Session` here: https://github.com/sddm/sddm/blob/b0306f3290e572be746df6586e39e778b2bdfb7c/src/common/Configuration.h#L105-L106

It's part of a slightly bigger patch: https://build.opensuse.org/package/view_file/openSUSE:Factory/sddm/0001-Read-the-DISPLAYMANAGER_AUTOLOGIN-value-from-sysconf.patch?expand=1&rev=79193c1cf4c386e10101fe881b6ad15d

**Comment by Batcastle** [comment]
Has this been implemented yet? I'm wanting to have the OS I develop, Drauger OS, ship with KDE for the next release and default to Wayland. This is working fine when auto-login is enabled. But when it's not enabled, SDDM defaults to the X11 KDE session. Just wanting to see if I should bother trying to figure out how to add this or not.
